### ISSUE TITLE
[Research Evaluation] Research Quality Observatory — external quality monitoring, hidden challenges, and cross-domain evaluation

### ISSUE THREAD

**Issue by FlapPearLabs** [FlapPearLabs/zhihu-grabber-toolkit#79]
```text
STATUS = DESIGN_CANDIDATE
IMPLEMENTATION_AUTHORIZATION = NONE
CURRENT_P1_DAG_MEMBER = NO
RELATION_TO_P1 = POST-P1 EVALUATION / OBSERVABILITY LAYER
```

# Concern

P1 Cross-Question Deep Research can eventually prove that the **system executes according to its declared contracts**:

- multi-query / multi-provider retrieval really ran;
- selected corpus identity is explicit;
- selected sources were actually analyzed;
- 100% Analysis Coverage is asserted only when selected/analyzed identities mechanically match;
- no silent provider/runtime fallback occurred;
- final claims are traceable to persisted artifacts;
- canonical runtime / model / run identity are auditable.

Those properties are necessary, but they do **not** answer a different and more important product-quality question:

> How good is the research itself in the open world, across many domains, and how far might the current baseline still be from a materially better system?

A pipeline can be internally correct, CI-green, fully traceable, and still:

- search too narrowly;
- miss important source groups;
- over-represent redundant viewpoints;
- stop retrieval too early;
- under-represent expert / minority / contradictory evidence;
- select sources that look semantically relevant but are not decision-useful;
- produce a polished synthesis that hides upstream recall defects;
- work well in AI/technology tasks but poorly in economics, medicine, history, law, education, or other domains.

Therefore:

```text
PIPELINE_CORRECTNESS != RESEARCH_QUALITY
100_PERCENT_ANALYSIS_COVERAGE != 100_PERCENT_RETRIEVAL_COVERAGE
GOOD_FINAL_WRITING != GOOD_RESEARCH
SELF_EVALUATION != EXTERNAL_QUALITY_EVIDENCE
```

The current concern is not that P1 is necessarily bad. The concern is that, even after P1 acceptance passes, its **true research quality may remain largely invisible** without an independent external measurement system.

# Why this should be a separate system

This issue does NOT propose changing the P1 retrieval/selection/analysis algorithms directly.

Instead, it proposes a future **Research Quality Observatory** around the product:

```text
User Research Task
      |
      v
Zhihu Deep Research Pipeline
      |
      +--> retrieval artifacts
      +--> source-group artifacts
      +--> selected corpus
      +--> claims / synthesis
      +--> final answer
      |
      v
Research Quality Observatory
      |
      +--> mechanical metrics
      +--> external/shadow research comparison
      +--> hidden-target challenges
      +--> metamorphic tests
      +--> sampled human/expert audits
      |
      v
Quality Evidence / Regression / Optimization Decisions
```

The observatory must remain logically separate from the main research controller. It measures and challenges the system; it does not silently change product decisions during the same run.

# Core design principle

Do NOT collapse research quality into one opaque score such as:

```text
RESEARCH_QUALITY = 87.3
```

That would hide failure location and encourage benchmark gaming.

Prefer a vector of orthogonal metrics that can identify **where** the system is weak.

Potential dimensions:

```text
Retrieval discovery / recall
Source-group coverage
Source diversity
Redundancy / duplication
Minority-view preservation
Expert-source coverage
Contradiction preservation
Freshness / temporal coverage
Selected->Analyzed coverage integrity
Claim support / citation entailment
Unsupported-claim rate
Cross-source synthesis faithfulness
Stability under equivalent prompts
Cost / latency / marginal information gain
```

# Proposed evaluation layers

## 1. Mechanical execution-quality metrics

These should be derived from existing structured artifacts where possible.

Examples:

- retrieval rounds executed;
- candidates discovered per round;
- marginal-new-candidate curve;
- provider contribution / overlap;
- source-group concentration;
- selected-content concentration;
- near-duplicate ratio;
- selected/analyzed set identity equality;
- evidence-per-claim distribution;
- contradiction count preserved into synthesis;
- cost / token / latency per research task.

These metrics are useful but **cannot reveal evidence the system never discovered**.

Therefore they are necessary, not sufficient.

## 2. Independent Shadow Researcher

For a sampled benchmark task, launch a separate research process that does NOT consume the Zhihu system's selected sources or source-group decisions.

Possible external evidence channels may include, subject to future authority and privacy policy:

- general web search;
- official sources;
- papers / standards / primary documents;
- external news / public databases;
- other independent research agents.

Compare discovered aspects / source groups / claims rather than only comparing final prose.

Example:

```text
ZHihu system groups = {A, B, C, D}
Shadow groups       = {A, B, C, D, E, F}

SHADOW_ONLY_GROUPS = {E, F}
```

`SHADOW_ONLY_GROUPS` are NOT automatically proof that P1 is wrong. They are **blind-spot candidates** requiring adjudication.

Repeated blind spots across tasks/domains become evidence of a structural weakness.

## 3. Hidden Target / Canary Retrieval Evaluation

Build curated tasks containing known-but-hidden high-value targets.

Examples of hidden targets:

- a low-popularity but highly technical answer;
- an important minority viewpoint;
- a specific question/source-group carrying a known counterargument;
- a primary source that should be discoverable under a reasonable query family.

The product is NOT told the targets during execution.

Measure:

```text
KNOWN_TARGET_RECALL
KNOWN_TARGET_GROUP_RECALL
KNOWN_TARGET_CLAIM_RECALL
```

This gives an approximate external retrieval-recall signal that final-answer quality alone cannot provide.

Important: hidden targets are not a claim of exhaustive ground truth. They are controlled probes.

## 4. Metamorphic / Stability Testing

Research systems often have no single gold answer. Metamorphic testing can detect instability without requiring full ground truth.

Take semantically equivalent task variants:

- paraphrase;
- reorder clauses;
- add/remove neutral context;
- Chinese vs English formulation where semantically equivalent;
- replace terminology with a recognized synonym;
- change wording without changing research intent.

Then compare:

- source-group overlap;
- core evidence overlap;
- claim overlap;
- conclusion consistency;
- retrieval depth;
- selected-corpus stability.

Potential metric families:

```text
CORE_GROUP_STABILITY
CORE_SOURCE_STABILITY
CLAIM_STABILITY
CONCLUSION_STABILITY
```

Large unexplained divergence is evidence of planner/retrieval/selection fragility.

## 5. Cross-domain stratified benchmark

A single aggregate benchmark is insufficient because different research domains fail differently.

Potential domain/task matrix:

| Domain | factual | contested | frontier | long-tail | temporal |
|---|---:|---:|---:|---:|---:|
| AI / software | ✓ | ✓ | ✓ | ✓ | ✓ |
| semiconductors | ✓ | ✓ | ✓ | ✓ | ✓ |
| economics / finance | ✓ | ✓ | ✓ | ✓ | ✓ |
| medicine / biotech | ✓ | ✓ | ✓ | ✓ | ✓ |
| education | ✓ | ✓ |  | ✓ | ✓ |
| history | ✓ | ✓ |  | ✓ |  |
| law / policy | ✓ | ✓ |  | ✓ | ✓ |
| consumer technology | ✓ |  | ✓ | ✓ | ✓ |

Metrics should be interpreted by task class.

For example:

- factual/current tasks emphasize precision, freshness, authority;
- contested questions emphasize viewpoint diversity and contradiction preservation;
- frontier questions emphasize novelty, recency, source diversity;
- long-tail questions emphasize recall of low-popularity but relevant evidence.

Do not force one evaluator rubric across every domain.

## 6. Stable + rotating + real-traffic evaluation sets

To avoid benchmark overfitting, use three layers:

```text
A. Stable Regression Set
   - fixed representative tasks
   - detects regressions after code / model / parameter changes

B. Rotating Hidden Challenge Set
   - regularly refreshed
   - hidden from optimization loop
   - reduces benchmark gaming

C. Sampled Real Research Tasks
   - sampled from actual use with appropriate privacy controls
   - detects failures missing from curated benchmarks
```

No single layer is sufficient by itself.

## 7. Limited human / expert audit

Do not require human review for every run.

A practical future model could be:

```text
100 runs
  -> 100 mechanical evaluations
  -> N automated independent/shadow evaluations
  -> 5-10 sampled human/expert audits
```

The most useful human question is not merely "is the answer good?"

Prefer:

> What important evidence, source group, counterargument, or domain-specific consideration did the system miss or distort?

Confirmed findings should be converted into durable regression assets:

```text
real defect
  -> hidden target / benchmark case / adversarial probe
  -> future automatic detection
```

# Candidate quality dashboard

A future evaluation record may expose metrics such as:

```text
RETRIEVAL_DISCOVERY
KNOWN_TARGET_RECALL
SHADOW_GROUP_COVERAGE
SOURCE_DIVERSITY
DUPLICATE_RATIO
GROUP_STARVATION_RISK
EXPERT_SOURCE_COVERAGE
CONTRADICTION_PRESERVATION
CLAIM_SUPPORT_RATE
UNSUPPORTED_CLAIM_RATE
COVERAGE_INTEGRITY
PROMPT_STABILITY
COST_PER_MATERIAL_DISCOVERY
```

The purpose is diagnostic observability, not leaderboard cosmetics.

Example:

```text
Final answer looks strong

KNOWN_TARGET_RECALL      = 0.61
SHADOW_GROUP_COVERAGE    = 0.54
DUPLICATE_RATIO          = 0.37
CLAIM_SUPPORT_RATE       = 0.96

Interpretation:
Downstream analysis/synthesis may be strong while upstream retrieval/selection is weak.
```

This is far more actionable than a single LLM score.

# Parameter-optimization policy candidate

This observatory should become the evidence source for future tuning.

Candidate rule:

```text
NO OBSERVABLE MATERIAL DEFECT
=> NO PARAMETER TUNING
```

Examples of parameters/strategies that must NOT be tuned merely because their current values look arbitrary:

- saturation threshold;
- max retrieval rounds;
- retrieval budget;
- novelty weight;
- MMR lambda;
- near-duplicate threshold;
- embedding model/profile;
- planner strategy;
- provider weighting.

Only consider tuning when external evaluation shows a repeatable, material failure pattern.

Example:

```text
Medicine hidden-target recall = 0.58
Other domains                 = 0.85+

Observed pattern:
medicine tasks repeatedly stop after round 1
shadow researcher consistently finds important missing groups

Then:
run a bounded experiment on saturation policy.
```

Any tuning experiment should have:

- a predeclared hypothesis;
- representative tasks;
- bounded candidate settings;
- quality + cost metrics;
- regression checks;
- a stop rule;
- no endless hyperparameter search.

# Relationship to existing backlog Issues

This proposal is intentionally distinct from existing product-enhancement candidates:

- #53 Gap-triggered Targeted Re-query
- #54 Pre-stop Escape Probe
- #55 Advanced Research-loop Enhancements

Those Issues describe potential changes to the **research algorithm/controller**.

This Issue describes the **measurement system required to decide whether such changes are actually needed**.

Conceptually:

```text
#53/#54/#55 = possible treatments
THIS ISSUE    = diagnostic / measurement system
```

Do not promote #53/#54/#55 merely because they sound theoretically useful. Their promotion should eventually be informed by observable quality evidence from a system like this one.

# Important non-goals

This design candidate does NOT authorize:

- changing current P1 selector parameters;
- changing embedding model/profile;
- adding new providers;
- adding targeted re-query;
- adding escape probes;
- adding RL/LTR;
- building a complex dashboard immediately;
- inventing a universal "research quality" scalar;
- treating another LLM's opinion as ground truth;
- building a huge benchmark before P1 itself is accepted.

# Suggested staged evolution

A future minimum viable observatory could be much smaller than the full design above.

Possible stages:

### Stage A — Evidence packet + mechanical metrics

Persist a structured evaluation packet per selected research run.

No algorithm changes.

### Stage B — Small cross-domain regression set

Create a limited but diverse task set with explicit task classes and several hidden targets.

### Stage C — Shadow research comparison

Run an independent researcher on a bounded sample and compare source groups / claims.

### Stage D — Metamorphic stability suite

Add paraphrase and equivalent-intent robustness tests.

### Stage E — sampled human/expert audit

Use real findings to expand the regression corpus.

Only after these stages produce evidence should parameter/model/controller optimization be considered.

# Promotion gate

This Issue is a design/evaluation backlog record only.

Do NOT implement before P1 is fully accepted and the post-P1 retrospective decides the correct scope.

Potential future promotion path:

```text
P1 canonical acceptance complete
-> post-P1 retrospective
-> evaluation design review
-> minimum viable observatory scope
-> independent review
-> formal ticket(s)
-> START_GATE
```

Before promotion:

```text
IMPLEMENTATION_AUTHORIZATION = NONE
CURRENT_P1_DAG_MEMBER = NO
P1_COMPLETION_BLOCKER = NO
```

# Open questions for future design review

1. What is the minimum representative cross-domain benchmark without turning this into a research project of its own?
2. Which metrics can be derived mechanically from existing P1 artifacts with near-zero incremental runtime cost?
3. How should shadow-research findings be adjudicated without treating the shadow system as ground truth?
4. How many hidden targets per task/domain are sufficient to detect structural recall problems?
5. How should evaluator-model bias and correlated model failures be controlled?
6. Which task classes require domain-expert review?
7. What is the acceptable evaluation cost relative to the normal research-run cost?
8. How should rotating hidden challenges be stored so the production agent cannot overfit them?
9. Which findings are strong enough to justify changing a P1 implementation default?
10. What quantitative/qualitative stop rule prevents the observatory itself from becoming an endless benchmark/tuning project?

# Current decision

```text
STATUS = DESIGN_CANDIDATE
IMPLEMENTATION_AUTHORIZATION = NONE
CURRENT_P1_DAG_MEMBER = NO
P1_COMPLETION_BLOCKER = NO
NEXT_REVIEW_POINT = AFTER_P1_CANONICAL_ACCEPTANCE_AND_POST_P1_RETROSPECTIVE
```

**Comment by FlapPearLabs** [comment]
PROJECT RETIREMENT CLOSURE

DISPOSITION = NOT_PLANNED
PROJECT_STATUS = ENDED
IMPLEMENTATION_STATUS = NOT_STARTED
REASON = PROJECT_ENDED_BEFORE_PROMOTION_OR_IMPLEMENTATION
HISTORICAL_TRUTH = Research Quality Observatory 是 POST-P1 评估 DESIGN_CANDIDATE，未获实施授权；其晋级门所依赖的最终 P1 canonical 验收未完成。项目结束前未晋级或实施，此归档不评价设计技术优劣。
NO_COMPLETION_CLAIM = YES
